app/models/merchant.rb

```diff
@@ -40,4 +40,29 @@ def deactivate_items
   def activate_items
     items.update(active?: true)
   end
+
+  def items_with_default_images
+    default = 'https://static.wixstatic.com/media/d8d60b_6ff8d8667db1462492d681839d85054c~mv2.png/v1/fill/w_900,h_900,al_c,q_90/file.jpg' 
+    items.where(image: default)
+  end
+
+  def unfulfilled_items_total
+    item_orders.where(status: 0).sum('item_orders.price * item_orders.quantity')
+  end
+
+  def unfulfilled_orders_count
+    item_orders.where(status: 0).count
+  end
+
+  def exceeds_inventory?(order)
+    if order.item_orders.joins(:item).where("items.merchant_id = #{self.id} AND item_orders.quantity > items.inventory") != []
```

Comment:
this one I use to check on an individual order and if that order's item_orders exceed and items quantity, generally how orders are constructed is that there is only one item_order per item per order, so simply checking the item order quantity against the item inventory works here